prince-of-versions/src/main/java/co/infinum/princeofversions/mvp/presenter/impl/PovPresenterImpl.java

```diff
@@ -37,24 +50,31 @@ public void checkForUpdates() {
         interactor.checkForUpdates(new PovInteractorListener() {
             @Override
             public void onUpdateAvailable(VersionContext version) {
-                // notify if there is no notification type or there was no notification before, or current version is not equal to last one.
-                String notificationType = version.getOptionalUpdate().getNotificationType();
-                String lastNotifiedVersion = repository.getLastVersionName(null);
-                boolean notNotifiedUpdateAvailable = lastNotifiedVersion == null || !lastNotifiedVersion
-                        .equals(version.getOptionalUpdate().getVersion()
-                                .getVersionString());
-                boolean alreadyNotifiedUpdateAvailable = lastNotifiedVersion != null && lastNotifiedVersion
-                        .equals(version.getOptionalUpdate().getVersion()
-                                .getVersionString());
-                if (notNotifiedUpdateAvailable || (alreadyNotifiedUpdateAvailable && notificationType != null && notificationType
-                        .equalsIgnoreCase("ALWAYS"))) {
-                    repository.setLastVersionName(version.getOptionalUpdate().getVersion().getVersionString());
-                    view.notifyOptionalUpdate(version.getOptionalUpdate().getVersion().getVersionString(), version.getMetadata());
+                if (checkIfUpdateShouldBeMade(version)) {
```

Comment:
Once changes in `PovInteractorImpl` are done I think this method won't be needed here anymore.